staging/src/k8s.io/cloud-provider/service/helpers/helper.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	utilnet "k8s.io/utils/net"
+)
+
+/*
+This file is duplicated from "k8s.io/kubernetes/pkg/api/v1/service/util.go"
```

Comment:
Tried to move some of the functions in `pkg/api/v1/service` and `pkg/api/service` here instead of duplicating it, but with the way they were used in internal registry / validation made it messy so just straight up duplicating the file seemed simpler.

Reply:
can't call methods here because we'd have to import internal APIs to external repos